feat: support nextjs v12

kirill-kruchkov opened the pull request:
NextJS v12 requires a `hostname` and `port` to be provided for the custom server when a middleware is used in the project (https://nextjs.org/docs/advanced-features/custom-server). Updating NextJS breaks such a project's dev server, if it uses `next-remote-watch@1.0.0`. Here's an example:

![image](https://user-images.githubusercontent.com/1702672/155010882-61843127-2a4c-4c1b-b948-5f29eb199be1.png)

This PR updates NextJS as a devDependency (not a peerDependency) of `next-remote-watch` and addresses the aforementioned issue by:
1. Reading a `HOSTNAME` alongside the `PORT` from env variables.
2. Supplying a default value for the hostname (`localhost`).
3. Providing both in the `next(...)` options.

It also adds a middleware to test this behaviour.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/next-remote-watch?pullRequest=35)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/next-remote-watch?pullRequest=35)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/next-remote-watch?pullRequest=35).</sub>

BRKalow commented:
Thanks for the contribution! It looks like it will, but just confirming, will this change work fine on older versions of next?

kirill-kruchkov commented:
Hello @BRKalow! I think it should, but let me test it again in the evening.

kirill-kruchkov commented:
@BRKalow I created a separate branch that includes this change while using Next v10. I haven't noticed anything unusual, but I'm not familiar well enough with all the test cases (there're 4 test scripts in the package.json). Would be great if someone could double check it: https://github.com/kirill-kruchkov/next-remote-watch/tree/test/next-10.

Also, I realised that this repo uses npm (I used yarn at first), so I updated this PR accordingly (dropped `yarn.lock` and updated `package-lock.json`).

BRKalow reviewed (commented):
Looks great, thanks!

MaxLeiter commented:
Will a new release be published with this merged?
